renku/cli/service.py

```diff
@@ -71,6 +85,93 @@ def run_worker(queues):
     start_worker(queues)
 
 
+def check_cmdline(cmdline, include=None):
+    """Check `cmdline` command of a process."""
+    include = include or []
+    service_components = include + SERVICE_COMPONENT_TAGS
+
+    for cmd in service_components:
+        if cmd in cmdline:
+            return True
+
+    return False
```

Comment:
Maybe it would be safer/more robust if we set an env variable, like `IS_RENKU=1` on the running subprocesses, then we could check for that with `psutil.Process(...).environ()`?

Reply:
I do see this as one way to achieve portability across different platforms, but I don't see why would that be more safer? 

To me it seems less safer cause environment variables of the process, could be, a mutable values, while the command is the signature of the process and as such is immutable. And even, if we do switch to this way we would still need to factor in for the service component tags and check the command line to determine which exact renku command we found, to exclude the self process. With that said, I believe current solution is fully portable and sufficiently safe (at least on the part of determining the process).

Reply:
They're not mutable outside the process from what I understand i.e. psutil only gives you the env variables as they were set when the process was started.

My concern was mostly that the current implementation would also pick up `python myscript.py --dataset api` or something like that, and with the env variables we could cut down on something like that happening. 

Not that there is a high chance of this, but if we spawn processes anyways and we can set their initial env variables easily, that is an easy to implement additional sanity check.